Add ActivityPub site selector and redirect handler

obenland opened the pull request:
Adds a canonical URL to select a site before redirecting to ActivityPub settings.
For Simple Sites it'll redirect to the Opt-In button (or settings), for all others either to plugin install or the settings.

The wordpress.com/social-web landing page will take advantage of it for its calls to action.

## Proposed Changes

* Add ActivityPub handler for `/marketing/activitypub` route that displays site selector
* Add redirect logic to ActivityPub settings page in wp-admin 
* Add plugin installation check for Jetpack/Atomic sites
* Redirect to plugin installation page if ActivityPub plugin not installed

## Why are these changes being made?

* Users need an easy way to access ActivityPub settings from Calypso
* The route should handle different site types (WordPress.com vs Jetpack/Atomic) appropriately
* For Jetpack sites, we need to check if the ActivityPub plugin is installed before redirecting to settings
* Provides a consistent user experience for managing ActivityPub settings across different site types

## Testing Instructions

* Navigate to `/marketing/activitypub` without a site selected → should show site selector
* Navigate to `/marketing/activitypub/your-wordpress-com-site` → should redirect to ActivityPub settings in wp-admin
* Navigate to `/marketing/activitypub/jetpack-site` with ActivityPub plugin installed → should redirect to ActivityPub settings  
* Navigate to `/marketing/activitypub/jetpack-site` without ActivityPub plugin → should redirect to plugin installation page with "activitypub" search term
* Test with user lacking `manage_options` capability → should show authorization error

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

obenland committed:
Add ActivityPub site selector and redirect handler

Adds a handler for /marketing/activitypub that displays a site selector
and redirects to the appropriate ActivityPub settings page. For Jetpack/Atomic
sites, checks if the ActivityPub plugin is installed and redirects to plugin
installation if needed.

obenland committed:
Merge branch 'trunk' into add/activitypub-site-selector

pfefferle commented:
The redirect for simple sites works perfect, but the `getPluginOnSite( state, siteId, 'activitypub' )` seem to not properly work. I tested with two Atomic sites, both have the plugin activated and I was always redirected to the plugin installation page.

obenland committed:
Fix ActivityPub controller to properly await plugin fetch

Ensures plugins are fetched and available before checking if ActivityPub plugin is installed for Jetpack/Atomic sites.

obenland committed:
Merge branch 'trunk' into add/activitypub-site-selector

obenland commented:
This should now work better with Jetpack and WoA sites

pfefferle reviewed (approved):
This works perfect! Awesome!
